src/Elasticsearch.Net.Aws/Elasticsearch.Net.Aws/AwsHttpConnection.cs

```diff
@@ -13,27 +12,45 @@ public class AwsHttpConnection : HttpConnection
     {
         readonly string _accessKey;
         readonly string _secretKey;
+        readonly string _token;
         readonly string _region;
 
         /// <summary>
-        /// Initializes a new instance of the AwsHttpConnection class.
+        /// Initializes a new instance of the AwsHttpConnection class with the specified AccessKey, SecretKey and Token.
         /// </summary>
         /// <param name="settings">The NEST/Elasticsearch.Net settings.</param>
         /// <param name="awsSettings">AWS specific settings required for signing requests.</param>
         public AwsHttpConnection(IConnectionConfigurationValues settings, AwsSettings awsSettings) : base(settings)
         {
             if (awsSettings == null) throw new ArgumentNullException("awsSettings");
+            if (string.IsNullOrWhiteSpace(awsSettings.AccessKey)) throw new ArgumentException("awsSettings.AccessKey is invalid.", "awsSettings");
             if (string.IsNullOrWhiteSpace(awsSettings.SecretKey)) throw new ArgumentException("awsSettings.SecretKey is invalid.", "awsSettings");
             if (string.IsNullOrWhiteSpace(awsSettings.Region)) throw new ArgumentException("awsSettings.Region is invalid.", "awsSettings");
             _accessKey = awsSettings.AccessKey;
             _secretKey = awsSettings.SecretKey;
+            _token = awsSettings.Token;
             _region = awsSettings.Region.ToLowerInvariant();
         }
 
+        /// <summary>
+        /// Initializes a new instance of the AwsHttpConnection class with info from the Instance Profile service
+        /// </summary>
+        /// <param name="settings">The NEST/Elasticsearch.Net settings.</param>
+        public AwsHttpConnection(IConnectionConfigurationValues settings, string region) : base(settings)
+        {
+            var credentials = InstanceProfileService.GetCredentials();
```

Comment:
If you need to periodically refresh the credentials you might want to call `GetCredentials()` in `SignRequest`. In many cases users will likely create a single client for the life-time of the application. In that situation you might get stuck with stale credentials.